Assets/Countly/Plugins/CountlySDK/Services/UserDetailsCountlyService.cs

```diff
@@ -487,6 +469,25 @@ private void AddToCustomData(string key, object value)
 
             CustomDataProperties.Add(key, value);
         }
+
+        /// <summary>
+        /// Checks if an internal call is providing valid key and has consent 
+        /// </summary>
+        /// <param name="key">property name</param>
+        private bool ValidateRequest(string key)
+        {
+            if (!_consentService.CheckConsentInternal(Consents.Users)) {
+                Log.Debug("[UserDetailsCountlyService] ValidateRequest, Consent is not given, ignoring the request.");
```

Comment:
and please also add function tag here rather than usage of ValidateRequest, by that way we could easily detect any anomaly log etc

Reply:
i will respect your opinion if you find what i did as overkill :D but will also ask your suggestion if so

Reply:
It was not overkill, it is like how I imagined, thanks for your rowk